### PR TITLE
AMS updates

### PR DESCRIPTION
**Pull request by marchdf**
**Pull-request type:**
- [X] Bug fix 
- [ ] Documentation update
- [X] Feature enhancement

## Description of the pull-request

This provides an update to AMS. The following changes were made:

- Updated Peclet calculation for AMS (uses the turbulent viscosity as opposed to the previous version which didnt)
- Fix inconsistency with the calculation of CM43scale (it used to be a slightly different calculation in two different parts of the code)
- Added user input for CMdeg
- Added the possibility of outputting the peclet number
- Added an AMS `post-iter` work to correctly output AMS forcing components
- Added an AMS `predict_state` to correctly handle AMS average quantities

This should introduce no diffs in regression tests _except_ AMS ones. We expect diffs there.

## Checklist

**NOTE** The checklist below is a suggestion and not mandatory. You don't have
to _check all boxes_ to submit a pull request. However, please add a brief
explanation in your pull request summary explaining the omission for the benefit
of reviewers and developers.

*All pull requests*
- [X] Builds successfully (*must test on at least one system/compiler combination*)
  - Operating systems 
    - [X] Linux
    - [ ] MacOS
  - Compilers 
    - [X] GCC
    - [ ] LLVM/Clang
    - [X] Intel compilers
    - [ ] NVIDIA CUDA
- [X] Compiles without warnings
- [ ] Passes all unit tests
- [ ] Passes all regression tests
- [ ] Documentation builds without errors

*For new code updates*
- [ ] Documentation updates - additions to user, theory, and verification manuals
- [ ] New unit tests providing coverage for new code, bug fixes etc.
- [ ] New regression tests exercising the new code

### CODE REVIEW

**Comment by jamelvin** [comment]
Note that this also:

- Adds a buffer to the division by wallDist in `src/node_kernels/MomentumSSTAMSForcingNodeKernel.C` to prevent `inf` errors
- Corrects an averaging bug in overset cases where the invalid state was contributing to the running average

File: src/AMSAlgDriver.C [hunk 1]
```diff
@@ -17,6 +17,17 @@
 #include "ngp_utils/NgpFieldManager.h"
 #include "ngp_algorithms/MetricTensorElemAlg.h"
 #include "stk_mesh/base/NgpMesh.hpp"
+#include <stk_mesh/base/BulkData.hpp>
+#include <stk_mesh/base/Field.hpp>
+#include <stk_mesh/base/FieldParallel.hpp>
+#include <stk_mesh/base/MetaData.hpp>
+
+//stk_mesh/base/fem
+#include <stk_mesh/base/BulkData.hpp>
```
Comment (jamelvin) [inline src/AMSAlgDriver.C:26]: Looks like we duplicated these

Comment (marchdf) [reply src/AMSAlgDriver.C:26]: good catch

File: src/edge_kernels/AMSMomentumEdgePecletAlg.C [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      for (int d = 0; d < ndim; ++d) {
+        const DblType dxj =
+          coordinates.get(nodeR, d) - coordinates.get(nodeL, d);
+        asq += av[d] * av[d];
+        axdx += av[d] * dxj;
+        udotx += 0.5 * dxj * (vrtm.get(nodeR, d) + vrtm.get(nodeL, d));
+      }
+
+      const DblType pecnum =
+        rhoIp * stk::math::abs(udotx) / (muIp + pecScale_ * mutIp + eps);
```
Comment (psakievich) [inline src/edge_kernels/AMSMomentumEdgePecletAlg.C:106]: What is the purpose of the scaling factor here?

**Review by psakievich — APPROVED** [review]
Looks good to me.

File: src/edge_kernels/AMSMomentumEdgePecletAlg.C (hunk 2 above)
Comment (jamelvin) [reply src/edge_kernels/AMSMomentumEdgePecletAlg.C:106]: It's pretty adhoc at the moment, basically to allow us to force central differencing in a larger part of the flow and we needed to be able to control it as the turbine case was showing some sensitivity to it.

We are looking into a more theoretically sound formulation to replace this in the future.

Comment (psakievich) [reply src/edge_kernels/AMSMomentumEdgePecletAlg.C:106]: :+1:

**Review by gantech — APPROVED** [review]

**Pull request merged**